### PR TITLE
UCT/IB: Disable mlx5 over exp verbs

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**
## What
Fix for https://github.com/openucx/ucx/issues/8100

### CODE REVIEW

File: src/uct/ib/configure.m4 [hunk 1]
```diff
@@ -230,7 +183,8 @@ AS_IF([test "x$with_ib" = "xyes"],
                     ], [[#include <infiniband/mlx5dv.h>]])])])
 
        AS_IF([test "x$has_res_domain" = "xyes" -a "x$have_cq_io" = "xyes" ], [], [
-               with_mlx5_hw=no])
+              AC_MSG_NOTICE([no DV bare-metal support])
+               with_mlx5_hw=no with_mlx5_dv=no])
```
Comment (yosefe) [inline src/uct/ib/configure.m4:187]: why changed?

File: test/gtest/uct/ib/test_rc.cc [hunk 2]
```diff
@@ -806,7 +806,7 @@ test_uct_iface_attrs::attr_map_t test_rc_iface_attrs::get_num_iov() {
         EXPECT_TRUE(has_transport("rc_verbs"));
         m_e->connect(0, *m_e, 0);
         uct_rc_verbs_ep_t *ep = ucs_derived_of(m_e->ep(0), uct_rc_verbs_ep_t);
-        uint32_t max_sge;
+        uint32_t max_sge = 0; // for gcc 10 -Og
```
Comment (gleon99) [inline test/gtest/uct/ib/test_rc.cc:809]: Why?

Comment (Artemy-Mellanox) [reply test/gtest/uct/ib/test_rc.cc:809]: Without it getting error. Looks like compiler bug. 
```
test/gtest/uct/ib/test_rc.cc:814:34: error: 'max_sge' may be used uninitialized in this function [-Werror=maybe-uninitialized]
         iov_map["am"]  = max_sge - 1; // 1 iov reserved for am header
```

File: src/uct/ib/configure.m4 [hunk 3]
```diff
@@ -235,6 +188,8 @@ AS_IF([test "x$with_ib" = "xyes"],
        AS_IF([test "x$with_mlx5_hw" = "xyes"],
              [AC_MSG_NOTICE([Compiling with mlx5 bare-metal support])
               AC_DEFINE([HAVE_MLX5_HW], 1, [mlx5 bare-metal support])
+              AS_IF([test "x$with_mlx5_dv" = "xyes"],
```
Comment (yosefe) [inline src/uct/ib/configure.m4:191]: can we remove also with_mlx5_hw ?

File: src/uct/ib/configure.m4 [hunk 4]
```diff
@@ -132,58 +132,12 @@ AS_IF([test "x$with_ib" = "xyes"],
...
-           [verbs_exp=no])
-
-       AC_CHECK_MEMBERS([struct ibv_exp_device_attr.exp_device_cap_flags,
-                         struct ibv_exp_device_attr.odp_caps,
-                         struct ibv_exp_device_attr.odp_caps.per_transport_caps.dc_odp_caps,
-                         struct ibv_exp_device_attr.odp_mr_max_size,
-                         struct ibv_exp_qp_init_attr.max_inl_recv,
-                         struct ibv_async_event.element.dct],
-                        [], [], [[#include <infiniband/verbs_exp.h>]])
+       verbs_exp=no
```
Comment (yosefe) [inline src/uct/ib/configure.m4:135]: can we remove this var?

**Review by yosefe — DISMISSED** [review]

**Comment by yosefe** [comment]
pls squash and rebase

**Review by yosefe — APPROVED** [review]

**Pull request merged**